// 已废弃
syntax = "proto3";
message dm_V2 {
             int64 dmid        =  1; // 1=10
            string dmid_unique =  2; // f"{roomid}{dmid}"
            uint32 mode        =  3; // 弹幕类型 1:顶部正向滚动弹幕 4:底部弹幕_舰长_ 5?:顶部弹幕_总督_
            uint32 fontsize    =  4; // 弹幕字号
            uint32 color       =  5; // 弹幕颜色
            string mid_hash    =  6; // 发送着mid hash
            string content     =  7; // 弹幕正文
            uint64 ctime       =  8; // 发送时间TimeStamp(毫秒)
    repeated bytes unknown_09  =  9; // 
             int64 dmid_2      = 10; // 1=10
    repeated bytes unknown_11  = 11; // 
    repeated bytes unknown_12  = 12; // 
       chat_bubble chat_bubble = 13; // 聊天气泡
            uint32 dm_type     = 14; // 1:表情 2: 语音
    repeated emots emots       = 15; // 表情
    repeated bytes unknown_16  = 16; // 
    repeated bytes unknown_17  = 17; // 
           lottery lottery     = 18; // 抽奖
}
message emots{
    string desc = 1;
    emots_detail detail = 2;
    repeated bytes unknown_XXX_03 =  3;
    repeated bytes unknown_XXX_04 =  4;
}
message emots_detail{
    string emoticon_unique = 1;
    string url = 2;
    bool in_player_area = 3;
    bool is_dynamic = 4;
    bytes unknown_emots_detail_0005 = 5;
    uint32 height = 6;
    uint32 width = 7;
}
message lottery{
    bytes unknown_lottery_0001 = 1;
    uint32 is_lottery = 2;
    string lottery_id = 3;
}
message chat_bubble{
    uint32 chat_bubble_type  = 1;
    string chat_bubble_color = 2;
}